xarray/core/weighted.py

```diff
@@ -239,6 +290,81 @@ def _weighted_std(
 
         return cast("DataArray", np.sqrt(self._weighted_var(da, dim, skipna)))
 
+    def _weighted_quantile(
+        self,
+        da: "DataArray",
+        q,
+        dim: Optional[Union[Hashable, Iterable[Hashable]]] = None,
+        skipna: Optional[bool] = None,
+        keep_attrs: Optional[bool] = None,
+    ) -> "DataArray":
+        """Apply a weighted ``quantile`` to a DataArray along some dimension(s)."""
+
+        def _weighted_quantile_type7_1d(a, weights, q, skipna):
+            # This algorithm has been adapted from:
+            #   https://aakinshin.net/posts/weighted-quantiles/#reference-implementation
+            if skipna:
+                # Remove nans from a and weights
+                not_nan = ~np.isnan(a)
+                a = a[not_nan]
+                weights = weights[not_nan]
+            elif np.isnan(a).any():
+                return np.full(len(q), np.nan)
+            # Flatten input values because this function is 1d
+            a = a.ravel()
+            weights = weights.ravel()
+            n = a.size
+            assert n == len(weights)
+            weights = weights / weights.sum()
+            sorter = np.argsort(a)
```

Comment:
Will this sort work even if the arrays have been chunked with dask? 
Maybe add a dask test with small chunks?